### PR TITLE
Use callbacks instead of Asio for OrbitApp-to-OrbitMainWindow communication, stop using GTcpServer

### PR DESCRIPTION
**Pull request by dpallotti**

### CODE REVIEW

File: OrbitGl/LogDataView.cpp [hunk 1]
```diff
@@ -134,7 +134,7 @@ void LogDataView::OnContextMenu(const std::string& a_Action, int a_MenuIndex,
                                 const std::vector<int>& a_ItemIndices) {
   if (m_SelectedCallstack &&
       static_cast<int>(m_SelectedCallstack->m_Depth) > a_MenuIndex) {
-    GOrbitApp->GoToCode(m_SelectedCallstack->m_Data[a_MenuIndex]);
+    // Not implemented.
```
Comment (dimitry-) [inline OrbitGl/LogDataView.cpp]: FATAL maybe?

File: OrbitQt/orbitmainwindow.cpp [hunk 2]
```diff
@@ -128,6 +128,16 @@ OrbitMainWindow::OrbitMainWindow(QApplication* a_App,
       [this] { ui->RightTabWidget->setCurrentWidget(ui->LiveTab); });
   GOrbitApp->AddDisassemblyCallback(
       [this](const std::string& disassembly) { OpenDisassembly(disassembly); });
+  GOrbitApp->AddErrorMessageCallback(
+      [this](const std::string& title, const std::string& text) {
+        QMessageBox::critical(this, QString::fromStdString(title),
+                              QString::fromStdString(text));
+      });
+  GOrbitApp->AddInfoMessageCallback(
```
Comment (dimitry-) [inline OrbitQt/orbitmainwindow.cpp:133]: are they singular callbacks? maybe replace add with set and remove loops?

File: OrbitGl/LogDataView.cpp (hunk 1 above)
Comment (dpallotti) [reply OrbitGl/LogDataView.cpp]: I ended up removing the entire class in a later commit.

File: OrbitQt/orbitmainwindow.cpp (hunk 2 above)
Comment (dpallotti) [reply OrbitQt/orbitmainwindow.cpp:133]: The vast majority if not all of them are, yes. I was following the pattern of the previous code. I'll switch to single callbacks.

**Comment by dpallotti** [comment]
> out of curiosity - can we remove CoreApp now? or is it still needed?

Almost... There are still a couple of places where it's still used. I still have quite some cleanup to do after this pull request so we might get there soon.

File: OrbitQt/orbitmainwindow.cpp (hunk 2 above)
Comment (dpallotti) [reply OrbitQt/orbitmainwindow.cpp]: Done.

**Pull request merged**